crates/wasi-http/src/types_impl.rs

```diff
@@ -1,38 +1,44 @@
+use crate::bytestream::ByteStream;
+use crate::common::stream::TableStreamExt;
 use crate::poll::Pollable;
-use crate::r#struct::ActiveRequest;
+use crate::r#struct::{ActiveFields, ActiveRequest, HttpFields, HttpRequest, TableExt};
 use crate::types::{
     Error, Fields, FutureIncomingResponse, Headers, IncomingRequest, IncomingResponse,
     IncomingStream, Method, OutgoingRequest, OutgoingResponse, OutgoingStream, ResponseOutparam,
     Scheme, StatusCode, Trailers,
 };
-use crate::WasiHttp;
+use crate::WasiHttpCtx;
 use anyhow::{anyhow, bail};
-use std::collections::HashMap;
 
-impl crate::types::Host for WasiHttp {
+fn convert(error: crate::common::Error) -> anyhow::Error {
+    // if let Some(errno) = error.downcast_ref() {
+    //     Error::UnexpectedError(errno.to_string())
+    // } else {
+    error.into()
+    // }
+}
+
+impl crate::types::Host for WasiHttpCtx {
     fn drop_fields(&mut self, fields: Fields) -> wasmtime::Result<()> {
-        self.fields.remove(&fields);
+        self.table_mut().delete_fields(fields).map_err(convert)?;
         Ok(())
     }
     fn new_fields(&mut self, entries: Vec<(String, String)>) -> wasmtime::Result<Fields> {
-        let mut map = HashMap::new();
-        for item in entries.iter() {
+        let mut map = ActiveFields::new();
+        for (key, value) in entries.iter() {
             let mut vec = std::vec::Vec::new();
-            vec.push(item.1.clone());
-            map.insert(item.0.clone(), vec);
+            vec.push(value.clone());
+            map.insert(key.clone(), vec);
```

Comment:
Aren't these `clone`s redundant? If you didn't call `iter()` you could just move these instead, since the vector is moved into this function, right?